public/containers/RefreshAction/index.tsx

```diff
@@ -0,0 +1,152 @@
+          filterBlockedItems<DataStream>(
+            services,
+            selectedItems as DataStream[],
+            IndexOpBlocksType.Closed,
+            dataStreamBlockedPredicate
+          ).then((filteredDataStreamsResult) => {
+            setUnBlockedItems(filteredDataStreamsResult.unBlockedItems.map((item) => item.name));
+            setBlockedItems(filteredDataStreamsResult.blockedItems.map((item) => item.name));
+          });
+          break;
+      }
+    } else {
+      setUnBlockedItems([]);
+      setBlockedItems([]);
+    }
+  }, [visible, services]);
+
+  const onConfirm = useCallback(async () => {
+    if (!!services) {
+      const result = await services.commonService.apiCaller({
+        endpoint: "indices.refresh",
+        data: {
+          index: unBlockedItems.join(","),
+        },
+      });
+      if (result && result.ok) {
+        let message = `Refresh ${type} [${unBlockedItems.join(",")}] successfully`;
+        if (selectedItems.length === 0) {
+          message = `Refresh all open indexes successfully`;
+        }
+        coreServices.notifications.toasts.addSuccess(message);
+        onClose();
+      } else {
+        coreServices.notifications.toasts.addDanger(result?.error || "");
+      }
+    }
+  }, [unBlockedItems, services, coreServices]);
+
+  if (!visible) {
+    return null;
+  }
+
+  return (
+    <EuiModal onClose={onClose}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>Refresh {type}</EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <div style={{ lineHeight: 1.5 }}>
+          {selectedItems.length === 0 && (
+            <>
+              <p>All open indexes will be refreshed.</p>
+            </>
+          )}
+          {!!unBlockedItems.length && (
+            <>
+              <p>The following {type} will be refreshed.</p>
+              <ul style={{ listStyleType: "disc", listStylePosition: "inside" }}>
+                {unBlockedItems.map((item) => (
+                  <li key={item} data-test-subj={`UnblockedItem-${item}`}>
+                    {item}
+                  </li>
+                ))}
+              </ul>
+            </>
+          )}
+          <EuiSpacer />
+          {!!blockedItems.length && (
+            <>
+              <EuiCallOut color="warning" iconType="alert" title={`The following ${type} will not be refreshed because they are closed.`} />
+              <p />
+              <ul style={{ listStyleType: "disc", listStylePosition: "inside" }}>
+                {blockedItems.map((item) => (
+                  <li key={item} data-test-subj={`BlockedItem-${item}`}>
+                    {item}
+                  </li>
+                ))}
+              </ul>
+            </>
+          )}
+        </div>
+      </EuiModalBody>
+
+      <EuiModalFooter>
+        <EuiButtonEmpty onClick={onClose}>Cancel</EuiButtonEmpty>
+        <EuiButton data-test-subj="refreshConfirmButton" onClick={onConfirm} fill>
```

Comment:
We'd better disable this button if there is no unBlockedItems neither refresh all indices. For example, user select a blocked index, and he still click this button, we will call the api to refresh all indices

Reply:
Good point